tests/jobs/localisation/test_configs/test_field_config.py

```diff
@@ -0,0 +1,220 @@
+)
+def test_gaussian_config_valid_angle(azimuth, config_class, method):
+    config = config_class(
+        method=method, main_range=0.1, perp_range=0.1, azimuth=azimuth
+    )
+    assert config.azimuth == azimuth
+
+
+@pytest.mark.parametrize(
+    "config_class, method",
+    [(GaussianConfig, "gaussian_decay"), (ExponentialConfig, "exponential_decay")],
+)
+@pytest.mark.parametrize(
+    "azimuth, expected_error",
+    [
+        (-0.0001, "ensure this value is greater than or equal to 0.0"),
+        (360.0001, "ensure this value is less than or equal to 360"),
+    ],
+)
+def test_invalid_angle(config_class, method, azimuth, expected_error):
+    with pytest.raises(pydantic.ValidationError, match=expected_error):
+        config_class(method=method, main_range=0.1, perp_range=0.1, azimuth=azimuth)
+
+
+@pytest.mark.parametrize("decay_method", ["gaussian_decay", "exponential_decay"])
+def test_field_config_init(decay_method):
+    field_config = {
+        "method": decay_method,
+        "main_range": 0.1,
+        "perp_range": 0.1,
+        "azimuth": 10,
+    }
+    obs_config = {
+        "add": "*",
+    }
+    param_group = {"add": "*"}
+    config = CorrelationConfig(
+        name="a_name",
+        obs_group=obs_config,
+        param_group=param_group,
+        field_scale=field_config,
+        ref_point=[10, 10],
+        obs_context=["a", "b", "c"],
+        params_context=["a", "b", "c"],
+    )
+    assert config.field_scale.method == decay_method
+
+
+@pytest.mark.parametrize(
+    "field_config, surface_config",
+    [
+        pytest.param(
+            {
+                "method": "gaussian_decay",
+                "main_range": 0.1,
+                "perp_range": 0.1,
+                "azimuth": 10,
+            },
+            None,
+            id="Field scale, not surface scale",
+        ),
+        pytest.param(
+            None,
+            {
+                "method": "gaussian_decay",
+                "main_range": 0.1,
+                "perp_range": 0.1,
+                "azimuth": 10,
+                "surface_file": "surface_file.txt",
+            },
+            id="Surface scale, not field scale",
+        ),
+        pytest.param(
+            {
+                "method": "gaussian_decay",
+                "main_range": 0.1,
+                "perp_range": 0.1,
+                "azimuth": 10,
+            },
+            {
+                "method": "gaussian_decay",
+                "main_range": 0.1,
+                "perp_range": 0.1,
+                "azimuth": 10,
+                "surface_file": "surface_file.txt",
+            },
+            id="Field scale and surface scale",
+        ),
+    ],
+)
+def test_correlation_config_ref_point(field_config, surface_config):
+    # Create a surface file (should only exist)
+    with open("surface_file.txt", "w", encoding="utf-8") as fout:
+        fout.write("A test file")
```

Comment:
You dont really need to create a file here, you could just patch it:
```suggestion
import os
from unittest.mock import MagicMock
   
def test_correlation_config_no_ref_point(field_config, surface_config, monkeypatch):
    monkeypatch.setattr(os.path, "exists", MagicMock())
    ...
```

Reply:
The same applies below.

Reply:
Need some more explanation. Where is monkeypatch defined for instance?

Reply:
Documentation is here: https://docs.pytest.org/en/latest/how-to/monkeypatch.html

Reply:
OK